api-processor/src/main/java/com/synopsys/integration/alert/processor/api/extract/model/project/ComponentUpgradeGuidance.java

```diff
@@ -0,0 +1,37 @@
+/*
+ * api-processor
+ *
+ * Copyright (c) 2021 Synopsys, Inc.
+ *
+ * Use subject to the terms and conditions of the Synopsys End User Software License and Maintenance Agreement. All rights reserved worldwide.
+ */
+package com.synopsys.integration.alert.processor.api.extract.model.project;
+
+import java.util.Optional;
+
+import org.jetbrains.annotations.Nullable;
+
+import com.synopsys.integration.alert.common.message.model.LinkableItem;
+
+public class ComponentUpgradeGuidance {
+    private final LinkableItem shortTermUpgradeGuidance;
+    private final LinkableItem longTermUpgradeGuidance;
+
+    public static ComponentUpgradeGuidance none() {
+        return new ComponentUpgradeGuidance(null, null);
+    }
+
+    public ComponentUpgradeGuidance(@Nullable LinkableItem shortTermUpgradeGuidance, @Nullable LinkableItem longTermUpgradeGuidance) {
+        this.shortTermUpgradeGuidance = shortTermUpgradeGuidance;
+        this.longTermUpgradeGuidance = longTermUpgradeGuidance;
+    }
+
+    public Optional<LinkableItem> getShortTermUpgradeGuidance() {
+        return Optional.ofNullable(shortTermUpgradeGuidance);
+    }
+
+    public Optional<LinkableItem> getLongTermUpgradeGuidance() {
+        return Optional.ofNullable(longTermUpgradeGuidance);
+    }
```

Comment:
I would much rather have `ComponentUpgradeGuidance` be optional on a component than have its fields be optional (if possible).

Reply:
I don't believe it would make sense in this case since the short term and long term guidance could be optional. For example, it's possible there is no short term guidance but there is long term guidance or vice versa.